wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/repeater/data/table/NavigatorLabel.java

```diff
@@ -46,11 +46,21 @@
 	public NavigatorLabel(final String id, final IPageableItems pageable)
 	{
 		super(id);
-		setDefaultModel(new StringResourceModel("NavigatorLabel", this,
+		setDefaultModel(new StringResourceModel(getNavigationLabelKey(), this,
 			new Model<>(new LabelModelObject(pageable))));
 	}
 
-	private static class LabelModelObject implements IClusterable
+	/**
+	 * Allows to override the key used for navigation label.
+	 *
+	 * @return The i18n key.
+	 */
+	protected String getNavigationLabelKey()
```

Comment:
What is the problem you try to solve here ?
You know that you could be specific in your i18n files to use the component path in the tree to specify different labels for different usages ?

Reply:
I know all of the above. 1) we use a a single i18n file at APP level (not my decision) 2) path is error prone and this is part of a component used in many places in app.

Reply:
What bothers me here is that we don't do this anywhere else. It just feels inconsistent.

Reply:
I can revert this if we do the model object for navigation class public.

Reply:
reverted this change. It is enough for my use case to have **LabelModelObject** public instead of private.